libnavigation-ui/src/main/java/com/mapbox/navigation/ui/map/MapWayName.java

```diff
@@ -79,17 +79,27 @@ boolean removeOnWayNameChangedListener(OnWayNameChangedListener listener) {
   }
 
   void onStart() {
-    if (navigation != null) {
-      navigation.registerRouteProgressObserver(progressChangeListener);
-    }
+    registerObservers();
   }
 
   void onStop() {
     if (isTaskRunning()) {
       filterTask.cancel(true);
     }
+    unregisterObservers();
+  }
+
+  private void registerObservers() {
+    if (navigation != null) {
+      navigation.registerRouteProgressObserver(progressChangeListener);
```

Comment:
Should we rename `progressChangeListener`? It's not a _listener_ anymore but also it doesn't listen only for progress updates (`LocationObsever`). What do you think?

Reply:
absolutely